src/values/AbstractValue.js

```diff
@@ -445,7 +445,8 @@ export default class AbstractValue extends Value {
     left: Value,
     right: Value,
     loc?: ?BabelNodeSourceLocation,
-    kind?: string
+    kind?: string,
+    isTemporal: boolean = true
```

Comment:
A few things I don't like...
1) a non-? argument following ...? arguments.
2) I don't like the the default switching to having side effects; I think there was a general design to have all of the create... functions that don't have Temporal in their name being side-effect free.

Reply:
WHOA. From now on every binary op is temporal BY DEFAULT???? Not good. And very likely not intended.

The way to go here is for the caller to wrap the atemporal value in a temporal value whose build function just returns the atemporal value.

Reply:
Yea, I wasn't sure about this one so I just did this to communicate what change I wanted which was easier this way than forking the whole function.

I don't think wrapping is the right way to model this, because it is the op that is temporal, not the value it results in. I can see that leading to serialization bugs.

The serializer doesn't know that the value that the temporal depends on is actually temporal so it can serialize it where ever and then just refer to that value.

E.g. we'll want to wrap some of these temporals in `try/catch` and if the serializer decides to serialize the inner value separately, then the temporal value just refers to that you end up with:

```js
var value = left + right;
try {
  var temporalValue = value;
} catch (x) {
  ...
}
```

Reply:
Yea this was not intentional to make the default true. I switched from `isPure`->`isTemporal` and forgot to switch the default.